samples: display: lvgl: improve sample.yaml

gmarull opened the pull request:
- Remove redundant content
- Add filters based on DT so that board specifics can be removed and sample be made more generic at the same time
- Other minor fixes/additions, please check commits

gmarull committed:
samples: display: lvgl: adjust harness field

'display' harness doesn't exist, so replace it with 'none' to make
things more clear. It is required so that test is not run e.g. on CI,
test would timeout because the GUI is left ON forever.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
samples: display: lvgl: filter by zephyr,display chosen

Sample can be built on any board that defines a zephyr,display and has
it enabled.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: esp_wrover_kit: enable SPI if CONFIG_DISPLAY=y

If we enable DISPLAY=y, ILI9341 will be selected automatically if all of
its dependencies are met as well (SPI), let's add this to the board
defconfig.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
samples: display: lvgl: add min_flash/min_ram

LVGL eats a considerable amount of flash, so tests may fail on platforms
that meet all requirements but that have a low amount of flash, e.g.
nucleo_g071rb. Set min_flash for all cases to 250K (result of some quick
experimentation). Also update minimum ram to exclude platforms that
cannot run Zephyr + LVGL (e.g. those with just 16K).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: arm: bl5340_dvk: enable SPI if CONFIG_DISPLAY=y

ILI9341 display requires SPI bus, enable it if CONFIG_DISPLAY=y

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

brgl reviewed (dismissed):
Nice, I like these, approved.

gmarull committed:
samples: display: lvgl: remove extra samples

Simplify sample.yaml by just adding a single test case. In future
commits, test case will be refined to be based on DT filters so that it
is made more generic.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
modules: lvgl: disable for CPU_CORTEX_M0/M0PLUS

There is a compiler bug in the current Zephyr SDK (GCC 10.3.0) that
prevents using LVGL on ARM Cortex M0/M0+. Disable LVGL on such CPU until
the next SDK is deployed (GCC 12.1), as the issues does not appear
anymore.

Ref. https://github.com/lvgl/lvgl/issues/3425

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented on boards/arm/bl5340_dvk/Kconfig.defconfig:
```diff
@@ -106,5 +106,9 @@ default $(dt_compat_on_bus,$(DT_COMPAT_TI_TCA9538),i2c)
 
 endif # BOARD_BL5340_DVK_CPUAPP || BOARD_BL5340_DVK_CPUAPP_NS
 
+# enable SPI when enabling display drivers (for ILI9341)
+config SPI
+	default y if DISPLAY
+
```
nice catch, fixed
